smdebug/tensorflow/base_hook.py

```diff
@@ -339,6 +339,16 @@ def _close_writers(self) -> None:
         for device in to_delete_writers:
             del self.writer_map[device]
 
+        to_delete_writers = []
+        # Delete all the tb writers
+        for mode, writer in self.tb_writers.items():
+            if writer is not None:
+                writer.flush()
+                writer.close()
+                to_delete_writers.append(mode)
+        for mode in to_delete_writers:
+            del self.tb_writers[mode]
```

Comment:
Why do we need to explicitly delete these?

Reply:
all writers are closed and deleted in close_writers() marking the end of a file. initialize_writers() or maybe_get_tb_writer() creates a new writer for the next file - in this function, there is a check to see if a writer exists for a particular mode. if there is no writer for a mode, it creates a new one.
probably done this way (rather than just setting the writer to None) to avoid a scenario where a write is made using a None writer.